fdbserver/KeyValueStoreMemory.actor.cpp

```diff
@@ -462,39 +461,55 @@ class KeyValueStoreMemory final : public IKeyValueStore, NonCopyable {
 		return total;
 	}
 
-	// Data format for normal operation:
+	static bool isOpEncrypted(OpHeader* header) { return header->op >> 31 == 1; }
+
+	static void setEncryptFlag(OpHeader* header, bool set) {
+		if (set) {
+			header->op |= (1UL << 31);
+		} else {
+			header->op &= ~(1UL << 31);
+		}
+	}
+
+	// Unencrypted data format (1 byte padding at the end):
 	// +-------------+-------------+-------------+--------+--------+
 	// | opType      | len1        | len2        | param2 | param2 |
 	// | sizeof(int) | sizeof(int) | sizeof(int) | len1   | len2   |
 	// +-------------+-------------+-------------+--------+--------+
 	//
-	// However, if the operation is encrypted:
-	// +-------------+-------------+-------------+---------------------------------+-------------+--------+--------+
-	// | OpEncrypted | len1        | len2        | BlobCipherEncryptHeader         | opType      | param1 | param2 |
-	// | sizeof(int) | sizeof(int) | sizeof(int) | sizeof(BlobCipherEncryptHeader) | sizeof(int) | len1   | len2   |
-	// +-------------+-------------+-------------+---------------------------------+-------------+--------+--------+
-	// |                                plaintext                                  |           encrypted           |
-	// +-----------------------------------------------------------------------------------------------------------+
+	// Encrypted data format (1 byte padding at the end):
```

Comment:
Thanks for adding the comment. Please move it above, as this is common for `encrypted` and/or `unencrypted` log format.